Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -19,12 +19,11 @@ namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher
 {
     public class SearchSkimmer : Skimmer<AnalyzeContext>
     {
-        public const string AssetFingerprint = "AssetFingerprint/v1";
-        public const string ValidationFingerprint = "ValidationFingerprint/v1";
-        public const string ValidationFingerprintHash = "ValidationFingerprintHash/v1";
+        public const string UniqueSecretHashCurrent = "UniqueSecretHash256/current";
 
-        public const string AssetFingerprintV2 = "AssetFingerprint/v2";
-        public const string ValidationFingerprintV2 = "ValidationFingerprint/v2";
+        public const string AssetFingerprintCurrent = "AssetFingerprint/current";
+        public const string ValidationFingerprintCurrent = "ValidationFingerprint/current";
+        public const string UniqueSecretFingerprintCurrent = "UniqueSecretFingerprint/current";
```

Comment:
@michaelcfanning I'm sorry, but I am still not totally clear about this change based on versioning. In my mind, there are two distinct problems to solve:

1. Design a hash based on just the secret text which correlates with the hash generated by the legacy SPAM tool.

2. Extend the existing fingerprinting mechanism to account for changes in the fingerprint itself. 

I am unclear if this PR addresses (1).